pkg/storage/unified/sql/sqltemplate/dialect.go

```diff
@@ -184,3 +188,15 @@ type name string
 func (n name) DialectName() string {
 	return string(n)
 }
+
+// standardDerefString provides standard SQL of handling pointers to strings from go, dereferencing them if they have
+// a value, or returning NULL if not.
+// NOTE: This implementation doesn't work for comparison because this is done via "IS NULL", should be implemented separately
+type standardDerefString struct{}
+
+func (standardDerefString) DerefString(s *string) string {
+	if s == nil {
+		return "NULL"
+	}
+	return *s
```

Comment:
What is a difference between string "NULL" and nil pointer for which this function returns "NULL"? 🤔

Reply:
I was testing this yesterday in another branch (where I need to use this), and you are right, it still needs some changes to work correctly with the Arg function and handle this difference of NULL vs "NULL"